quodlibet/quodlibet/ext/events/synchronizedlyrics.py

```diff
@@ -0,0 +1,305 @@
+# -*- coding: utf-8 -*-
+# Synchronized Lyrics: a Quod Libet plugin for showing synchronized lyrics.
+# Copyright (C) 2015 elfalem
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of version 2 of the GNU General Public License as
+# published by the Free Software Foundation.
+
+
+"""Provides `Synchronized Lyrics` plugin for showing synchronized lyrics."""
+
+import os
+
+from datetime import datetime
+
+from gi.repository import Gtk, Gdk, GLib
+
+from quodlibet import app
+from quodlibet import qltk
+from quodlibet import config
+
+from quodlibet.plugins.events import EventPlugin
+
+
+class SynchronizedLyrics(EventPlugin):
+
+    PLUGIN_ID = 'SynchronizedLyrics'
+    PLUGIN_NAME = _('Synchronized Lyrics')
+    PLUGIN_DESC = _('Shows synchronized lyrics from .lrc file with same name \
+as the track.')
+
+    SYNC_PERIOD = 3000
+
+    _defaultBgColor = '#343428282C2C'
+    _defaultTxtColor = '#FFFFFFFFFFFF'
+    _defaultFontSize = 25
+
+    _lines = []
+    _timers = []
+
+    _currentLrc = ""
+
+    def PluginPreferences(self, window):
+        vb = Gtk.VBox(spacing=6)
+        vb.set_border_width(6)
+
+        t = Gtk.Table(n_rows=5, n_columns=2, homogeneous=True)
+        t.set_col_spacings(6)
+        t.set_row_spacings(3)
+
+        clrSection = Gtk.Label()
+        clrSection.set_markup("<b>" + _("Colors") + "</b>")
+        t.attach(clrSection, 0, 2, 0, 1)
+
+        l = Gtk.Label(label=_("Text:"))
+        l.set_alignment(xalign=1.0, yalign=0.5)
+        t.attach(l, 0, 1, 1, 2, xoptions=Gtk.AttachOptions.FILL)
+
+        c = Gdk.RGBA()
+        c.parse(self.getTextColor())
+        b = Gtk.ColorButton(rgba=c)
+        t.attach(b, 1, 2, 1, 2)
+        b.connect('color-set', self.setTextColor)
+
+        l = Gtk.Label(label=_("Background:"))
+        l.set_alignment(xalign=1.0, yalign=0.5)
+        t.attach(l, 0, 1, 2, 3, xoptions=Gtk.AttachOptions.FILL)
+
+        c = Gdk.RGBA()
+        c.parse(self.getBgColor())
+        b = Gtk.ColorButton(rgba=c)
+        t.attach(b, 1, 2, 2, 3)
+        b.connect('color-set', self.setBgColor)
+
+        fontSection = Gtk.Label()
+        fontSection.set_markup("<b>" + _("Font") + "</b>")
+        t.attach(fontSection, 0, 2, 3, 4)
+
+        l = Gtk.Label(label=_("Size (px):"))
+        l.set_alignment(xalign=1.0, yalign=0.5)
+        t.attach(l, 0, 1, 4, 5, xoptions=Gtk.AttachOptions.FILL)
+
+        a = Gtk.Adjustment.new(self.getFontSize(), 6, 36, 1, 3, 0)
+        s = Gtk.SpinButton(adjustment=a)
+        s.set_numeric(True)
+        s.set_text(str(self.getFontSize()))
+        t.attach(s, 1, 2, 4, 5)
+        s.connect('value-changed', self.setFontSize)
+
+        vb.pack_start(t, False, False, 0)
+        return vb
+
+    def getTextColor(self):
+        v = config.get("plugins", self.PLUGIN_ID + "_txt",
```

Comment:
There's actually a class to wrap all this (consistently). 
See [PluginConfigMixin](https://github.com/quodlibet/quodlibet/blob/master/quodlibet/quodlibet/plugins/__init__.py#L456).